hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/spi/impl/OzoneManagerServiceProviderImpl.java

```diff
@@ -381,25 +397,36 @@ connectionFactory, getOzoneManagerSnapshotUrl(),
    */
   @VisibleForTesting
   boolean updateReconOmDBWithNewSnapshot() throws IOException {
+    // Check permissions of the Recon DB directory
+    checkAndValidateReconDbPermissions();
     // Obtain the current DB snapshot from OM and
     // update the in house OM metadata managed DB instance.
     long startTime = Time.monotonicNow();
     DBCheckpoint dbSnapshot = getOzoneManagerDBSnapshot();
     metrics.updateSnapshotRequestLatency(Time.monotonicNow() - startTime);
-    if (dbSnapshot != null && dbSnapshot.getCheckpointLocation() != null) {
-      LOG.info("Got new checkpoint from OM : " +
-          dbSnapshot.getCheckpointLocation());
-      try {
-        omMetadataManager.updateOmDB(
-            dbSnapshot.getCheckpointLocation().toFile());
-        return true;
-      } catch (IOException e) {
-        LOG.error("Unable to refresh Recon OM DB Snapshot. ", e);
-      }
-    } else {
-      LOG.error("Null snapshot location got from OM.");
+
+    if (dbSnapshot == null) {
+      LOG.error("Failed to obtain a valid DB snapshot from Ozone Manager. This could be due to " +
+          "missing SST files or other fetch issues.");
+      return false;
+    }
+
+    if (dbSnapshot.getCheckpointLocation() == null) {
+      LOG.error("Snapshot checkpoint location is null, indicating a failure to properly fetch or " +
+          "store the snapshot.");
+      return false;
+    }
+
+    LOG.info("Attempting to update Recon OM DB with new snapshot located at: {}",
+        dbSnapshot.getCheckpointLocation());
+    try {
+      omMetadataManager.updateOmDB(dbSnapshot.getCheckpointLocation().toFile());
+      LOG.info("Successfully updated Recon OM DB with new snapshot.");
```

Comment:
I would recommend this to change to debug level. because this will flood the logs of recon. And also log the sequence number being used to fetch the OM DB snapshot. 
And check if this task details being updated in ReconTask table as suggested above with sequence number and list of SST files.